Address CRP review comments and clarify explicit/implicit rejection (#44) (#110)
cfrg/draft-irtf-cfrg-generic-hybrid-kems: draft-irtf-cfrg-hybrid-kems.md

Given the novelty of PQ algorithms, however, there is some concern that PQ
algorithms currently believed to be secure will be broken.  Hybrid
constructions that combine both PQ and traditional algorithms can help
moderate this risk while still providing security against quantum attacks.  If
constructed properly, a hybrid KEM will retain certain security properties
even if one of the two constituent KEMs is compromised, for whatever reason
that compromise might occur. If the PQ KEM is
broken, then the hybrid KEM should continue to provide security against
non-quantum attackers by virtue of its traditional KEM component. If the
traditional KEM is broken -- whether by a quantum computer running a polynomial-time attack or by some classical cryptanalytic advance -- then the hybrid KEM
should continue to resist quantum attacks by virtue of its PQ KEM component.
The advent of quantum computing lends particular urgency to the latter case,
but the hybrid construction guards against the failure of either component

> regardless of the cause.

Reviewer: ```suggestion
traditional KEM is broken -- whether by a quantum computer running a
polynomial-time attack or by some classical cryptanalytic advance -- then
the hybrid KEM should continue to resist quantum attacks by virtue of
its PQ KEM component. The advent of quantum computing lends particular
urgency to the latter case, but the hybrid construction guards against
the failure of either component regardless of the cause.
```

Given the novelty of PQ algorithms, however, there is some concern that PQ
algorithms currently believed to be secure will be broken.  Hybrid
constructions that combine both PQ and traditional algorithms can help
moderate this risk while still providing security against quantum attacks.  If
constructed properly, a hybrid KEM will retain certain security properties
even if one of the two constituent KEMs is compromised, for whatever reason
that compromise might occur. If the PQ KEM is
broken, then the hybrid KEM should continue to provide security against
non-quantum attackers by virtue of its traditional KEM component. If the
traditional KEM is broken -- whether by a quantum computer running Shor's

> algorithm or by some classical cryptanalytic advance -- then the hybrid KEM

Reviewer: ```suggestion
traditional KEM is broken -- whether by a quantum computer running a polynomial-time attack or by some classical cryptanalytic advance -- then the hybrid KEM
```

# Notation

> This document is consistent with all terminology defined in {{?RFC9794}}.  In

Reviewer: ```suggestion
This document is consistent with terminology defined in {{?RFC9794}}.  In
```

process.  Some implementations of component schemes do not support the
`DeriveKeyPair` function, only `GenerateKeyPair`. Likewise in the nominal
group case, a (scalar, group element) pair will only be generated when the
scalar is generated internally to the implementation.

An implementation of a hybrid KEM in such environments MAY deviate from the
above description in the following ways:

* `DeriveKeyPair` is not implemented.
* `GenerateKeyPair` returns a decapsulation key that carries both the
  decapsulation and encapsulation keys of each component, rather than a shared
  seed:

  ~~~
  dk_H = (dk_PQ, ek_PQ, dk_T, ek_T)
  ek_H = (ek_PQ, ek_T)
  ~~~

>   Each element may be a key or a pointer/handle to a key.  Carrying the

Reviewer: @bifurcation wdyt

```suggestion
  Each element can be a key or a pointer/handle to a key.  Carrying the
```

* Reference (optional): The document where this hybrid KEM is defined

To preserve the domain separation described in {{domain-separation}}, the set
of registered `Label` values MUST be suffix-free: no registered `Label` may be
a suffix of another.  A registration request whose `Label` is a suffix of an
already-registered `Label`, or of which an already-registered `Label` is a

> suffix, MUST be rejected.

Reviewer: I'm pretty sure the literature is explicitly about prefix-free, not suffix-free

<img width="585" height="85" alt="Image" src="https://github.com/user-attachments/assets/8e70c2ba-08f7-4278-9c4a-84f210e7c39d" />

Reviewer: 'yes but we are doing suffixes, not prefixes'
